### PR TITLE
added endpoint and tests for displaying a selected articles comments

### PR DESCRIPTION
**Pull request by oregan1**

### CODE REVIEW

File: model/model.js [hunk 1]
```diff
@@ -59,15 +59,24 @@ exports.selectArticles = () => {
     })
 }
 
-exports.selectArticlesComments = (id) => {
+exports.selectArticlesComments = async (id) => {
+    const articles = await db.query(`SELECT * FROM articles`)
     return db.query(`SELECT * FROM comments WHERE article_id = $1`,[id])
     .then((result) => {
+        const numArticles = articles.rows.length;
         result.rows[0];
```
Comment (emilyb93) [inline model/model.js:67]: Why do we have just this loose expression? What is it doing?

File: model/model.js [hunk 2]
```diff
@@ -59,15 +59,24 @@ exports.selectArticles = () => {
...
-        if (!result.rows[0]){ 
-            return Promise.reject({
-                status: 404,
-                msg: 'No article found with that id'
-            })
+        if (!result.rows[0]){
+            if (id <= numArticles){
+                return Promise.reject({
+                    status: 200,
+                    msg: 'No comments for this article'
```
Comment (emilyb93) [inline model/model.js:72]: This shouldnt be a rejection as technically there is no error here and you are sending a 200. From a users standpoint, they are expecting an array of comments. If there is no comments, we should be sending back an empty array for data consistency.

We're going to have to check if the article exists as well as checking for any comments

File: __tests__/app.test.js [hunk 3]
```diff
@@ -236,6 +236,15 @@ describe('app', () => {
                 })
             });
         });
+        test('if an article has no comments responds with no comments msg and status 200', () => {
+            const idToSearch = 10
+            return request(app)
+            .get(`/api/articles/${idToSearch}/comments`)
+            .expect(200)
+            .then(({body}) => {
+                expect(body.msg).toEqual('No comments for this article');
```
Comment (emilyb93) [inline __tests__/app.test.js:245]: We should expect an empty array, not a string. Data consistency is very important, if everything goes correctly, we should only return one type of data

**Review by emilyb93 — CHANGES_REQUESTED** [review]

File: model/model.js [hunk 4]
```diff
@@ -57,4 +55,18 @@ exports.selectArticles = () => {
     .then((result) => {
         return result.rows;
     })
+}
+
+exports.selectArticlesComments = async (id) => {
+    const articles = await db.query(`SELECT * FROM articles`)
+    return db.query(`SELECT * FROM comments WHERE article_id = $1`,[id])
+    .then((result) => {
+        if (!result.rows[0] && id > articles.rows.length){
```
Comment (two-triangles) [inline model/model.js:64]: We can just check whether articles.rows is empty, rather than using the id

**Comment by two-triangles** [comment]
Wonderful stuff! Merge away!

**Pull request merged**